core/imds/src/main/java/software/amazon/awssdk/imds/internal/DefaultEc2Metadata.java

```diff
@@ -130,7 +147,64 @@ public String toString() {
      */
     @Override
     public String get(String path) {
-        return "IMDS";
+
+        String data = null;
+        AbortableInputStream abortableInputStream = null;
+        try {
+            String token = getToken();
+            URI uri = URI.create(endpoint + path);
+            HttpExecuteRequest httpExecuteRequest = requestMarshaller.createDataRequest(uri , SdkHttpMethod.GET, token,
+                                                                                tokenTtl);
+            HttpExecuteResponse response =  httpClient.prepareRequest(httpExecuteRequest).call();
+            int statusCode = response.httpResponse().statusCode();
+
+            if (statusCode == HttpURLConnection.HTTP_OK) {
```

Comment:
Will it ever be something else in the 2xx family that we need to handle?

Reply:
According to the SEP, only 200 is mentioned. For other status codes, I have the else block to handle.